pkg/pull/signature.go

```diff
@@ -17,51 +16,65 @@ var (
 )
 
-type Signature struct {
+type InnerSignature struct {
 	LicenseSignature []byte `json:"licenseSignature"`
 	PublicKey        string `json:"publicKey"`
 	KeySignature     []byte `json:"keySignature"`
 }
 
+type OuterSignature struct {
+	LicenseData    []byte `json:"licenseData"`
+	InnerSignature []byte `json:"innerSignature"`
+}
+
 type KeySignature struct {
 	Signature   []byte `json:"signature"`
 	GlobalKeyId string `json:"globalKeyId"`
 }
 
-func VerifySignature(license *kotsv1beta1.License) error {
-	signature := &Signature{}
-	if err := json.Unmarshal(license.Spec.Signature, signature); err != nil {
+func VerifySignature(license *kotsv1beta1.License) (*kotsv1beta1.License, error) {
+	outerSignature := &OuterSignature{}
+	if err := json.Unmarshal(license.Spec.Signature, outerSignature); err != nil {
+		return nil, errors.Wrap(err, "failed to unmarshal license outer signature")
+	}
+
+	innerSignature := &InnerSignature{}
+	if err := json.Unmarshal(outerSignature.InnerSignature, innerSignature); err != nil {
 		// old licenses's signature is a single space character
-		if len(license.Spec.Signature) == 0 || len(license.Spec.Signature) == 1 {
-			return ErrSignatureMissing
+		if len(outerSignature.InnerSignature) == 0 || len(outerSignature.InnerSignature) == 1 {
+			return nil, ErrSignatureMissing
 		}
-		return errors.Wrap(err, "failed to unmarshal license signature")
+		return nil, errors.Wrap(err, "failed to unmarshal license inner signature")
 	}
 
 	keySignature := &KeySignature{}
-	if err := json.Unmarshal(signature.KeySignature, keySignature); err != nil {
-		return errors.Wrap(err, "failed to unmarshal key signature")
+	if err := json.Unmarshal(innerSignature.KeySignature, keySignature); err != nil {
+		return nil, errors.Wrap(err, "failed to unmarshal key signature")
 	}
 
 	globalKeyPEM, ok := publicKeys[keySignature.GlobalKeyId]
 	if !ok {
-		return errors.New("unknown global key")
+		return nil, errors.New("unknown global key")
 	}
 
-	if err := verify([]byte(signature.PublicKey), keySignature.Signature, globalKeyPEM); err != nil {
-		return errors.Wrap(err, "failed to verify key signature")
+	if err := verify([]byte(innerSignature.PublicKey), keySignature.Signature, globalKeyPEM); err != nil {
+		return nil, errors.Wrap(err, "failed to verify key signature")
 	}
 
-	licenseMessage, err := getMessageFromLicense(license)
-	if err != nil {
-		return errors.Wrap(err, "failed to convert license to message")
+	if err := verify(outerSignature.LicenseData, innerSignature.LicenseSignature, []byte(innerSignature.PublicKey)); err != nil {
+		return nil, errors.Wrap(err, "failed to verify license signature")
 	}
 
-	if err := verify(licenseMessage, signature.LicenseSignature, []byte(signature.PublicKey)); err != nil {
-		return errors.Wrap(err, "failed to verify license signature")
+	verifiedLicense := &kotsv1beta1.License{}
+	if err := json.Unmarshal(outerSignature.LicenseData, verifiedLicense); err != nil {
```

Comment:
I believe your current PR has `outerSignature.LicenseData` being base64'd here
One of the two will need to change (IMO `outerSignature.LicenseData` doesn't need base64, so you should remove it in vandoor)
Unless I'm misreading things

Reply:
Here it's defined as `[]byte` so it's automatically decoded.

Reply:
Updated in license-api